### PR TITLE
Let IHostBuilder AddNLog work correctly with ${configsetting}

### PR DESCRIPTION
**Pull request by snakefoot**
IHostBuilder AddNLog attempt to setup ConfigSettingLayoutRenderer early

Increase the chance that fluent configuration will work.

### CODE REVIEW

File: src/NLog.Web.AspNetCore/AspNetExtensions.cs [hunk 1]
```diff
@@ -112,8 +112,11 @@ public static LogFactory ConfigureNLog(this ILoggingBuilder builder, LoggingConf
         /// <param name="configFileName">Path to NLog configuration file, e.g. nlog.config. </param>
         public static ILoggingBuilder AddNLog(this ILoggingBuilder builder, string configFileName)
         {
-            ConfigureServicesNLog(null, builder.Services, serviceProvider => serviceProvider.GetService<IConfiguration>());
-            LogManager.LoadConfiguration(configFileName);
+            ConfigureServicesNLog(null, builder.Services, (s, c) =>
```
Comment (304NotModified) [inline src/NLog.Web.AspNetCore/AspNetExtensions.cs]: Please use longer names then `s` and `c`. This isn't C ;)

**Comment by 304NotModified** [comment]
> Increase the chance that fluent configuration will work.

Could you elaborate this a bit? Not sure what we are to improve here.

**Comment by snakefoot** [comment]
> Could you elaborate this a bit? Not sure what we are to improve here.

My initial version of AddNLog loaded the NLog-config before `IHostBuilder.Build()` had been called. Thus the registration code for setting up config (etc.) had not been called. Causing all Layout-renderers that depends on `${configsetting}` to fail.

By delaying the loading of NLog-config until `IHostBuilder.Build()` is called will help on this.

File: src/NLog.Web.AspNetCore/AspNetExtensions.cs (hunk 1 above)
Comment (snakefoot) [reply src/NLog.Web.AspNetCore/AspNetExtensions.cs]: Found a better way that got rid of short names.

**Comment by 304NotModified** [comment]
clear! Thanks!

**Pull request merged**